index.js

```diff
@@ -1 +1,151 @@
-module.exports = require('./lib/type');
+'use strict';
+/* globals Promise: false, Symbol: false */
+/* !
+ * type-detect
+ * Copyright(c) 2013 jake luer <jake@alogicalparadox.com>
+ * MIT Licensed
+ */
+var isPrototypeOfExists = typeof Object.getPrototypeOf === 'function';
+var promiseExists = typeof Promise === 'function';
+var globalObject = typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : self; // eslint-disable-line
+var isArrayExists = typeof Array.isArray === 'function';
+var symbolDoesNotExist = typeof Symbol === 'undefined';
+var toStringLeftSliceLength = 8;
+var toStringRightSliceLength = -1;
+/**
+ * ### typeOf (obj)
+ *
+ * Uses `Object.prototype.toString` to determine the type of an object,
+ * normalising behaviour across engine versions & well optimised.
+ *
+ * @param {Mixed} object
+ * @return {String} object type
+ * @api public
+ */
+module.exports = function typeDetect(obj) {
+  /* ! Speed optimisation
+   * Pre:
+   *   string literal     x 3,039,035 ops/sec ±1.62% (78 runs sampled)
+   *   boolean literal    x 1,424,138 ops/sec ±4.54% (75 runs sampled)
+   *   number literal     x 1,653,153 ops/sec ±1.91% (82 runs sampled)
+   *   null               x 8,448,576 ops/sec ±1.65% (59 runs sampled)
+   *   undefined          x 9,978,660 ops/sec ±1.92% (75 runs sampled)
+   *   function           x 2,556,769 ops/sec ±1.73% (77 runs sampled)
+   * Post:
+   *   string literal     x 38,564,796 ops/sec ±1.15% (79 runs sampled)
+   *   boolean literal    x 31,148,940 ops/sec ±1.10% (79 runs sampled)
+   *   number literal     x 32,679,330 ops/sec ±1.90% (78 runs sampled)
+   *   null               x 30,573,407 ops/sec ±0.58% (82 runs sampled)
+   *   undefined          x 32,363,368 ops/sec ±1.07% (82 runs sampled)
+   *   function           x 31,296,870 ops/sec ±0.96% (83 runs sampled)
+   */
+  var typeofObj = typeof obj;
+  if (typeofObj !== 'object') {
+    return typeofObj;
+  }
+
+  /* ! Spec Conformance
+   * ES5$15.2.4.2 (ES6$19.1.3.6) - Object.prototype.toString returns "[object Null]" for null object
+   * Test: `Object.prototype.toString.call(null)``
+   *  - PhantomJS 1.x == "[object DOMWindow]" (https://github.com/ariya/phantomjs/issues/13617)
+   *  - IE <=8  == "[object Object]"
+   */
+  if (obj === null) {
```

Comment:
TYVM for testing in Phantom1 - that is super helpful! IE8 fails on lots of the DOM stuff too. While the code will work for the most part in both of these engines, I think there's little point officially supporting them. We'd have to add a lot of code normalising all of the weird behaviours in such old engines - engines which are no longer supported by their vendors no less.

BTW I just realised I was being an idiot the whole time... the `if (obj === undefined)` would get hit _prior_ to benching and adding the hot `typeof path` on L42! We definitely don't need the undefined check now... silly me.

I'll keep `obj === null` because it benches much faster than going through the object calls. I'll make the modification and upload shortly.